### ISSUE TITLE
Can not use some plugins

### ISSUE THREAD

**Issue by waeljammal** [jwplayer/buildpipe#36]
Trying to nest plugins causes a representation error:

```
    buildkite:
      command:
        - docker-compose -f docker-compose-ci.yml build
      plugins:
        - docker-login#v2.0.1:
          - server: containers.amphoratech.net
            username: myUser
            password-env: MY_PASS
```

This will throw

```

Traceback (most recent call last):
--
  | File "/usr/bin/buildpipe", line 10, in <module>
  | sys.exit(main())
  | File "/usr/lib/python3.6/site-packages/buildpipe/__main__.py", line 17, in main
  | pipeline.create_pipeline(**vars(args))
  | File "/usr/lib/python3.6/site-packages/buildpipe/pipeline.py", line 206, in create_pipeline
  | steps.to_yaml(filename=outfile, Dumper=yaml.dumper.SafeDumper)
  | File "/usr/bin/box.py", line 733, in to_yaml
  | encoding=encoding, errors=errors, **yaml_kwargs)
  | File "/usr/bin/box.py", line 106, in _to_yaml
  | **yaml_kwargs)
  | File "/usr/lib/python3.6/site-packages/yaml/__init__.py", line 290, in dump
  | return dump_all([data], stream, Dumper=Dumper, **kwds)
  | File "/usr/lib/python3.6/site-packages/yaml/__init__.py", line 278, in dump_all
  | dumper.represent(data)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 27, in represent
  | node = self.represent_data(data)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 48, in represent_data
  | node = self.yaml_representers[data_types[0]](self, data)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 207, in represent_dict
  | return self.represent_mapping('tag:yaml.org,2002:map', data)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 118, in represent_mapping
  | node_value = self.represent_data(item_value)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 48, in represent_data
  | node = self.yaml_representers[data_types[0]](self, data)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 199, in represent_list
  | return self.represent_sequence('tag:yaml.org,2002:seq', data)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 92, in represent_sequence
  | node_item = self.represent_data(item)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 48, in represent_data
  | node = self.yaml_representers[data_types[0]](self, data)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 207, in represent_dict
  | return self.represent_mapping('tag:yaml.org,2002:map', data)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 118, in represent_mapping
  | node_value = self.represent_data(item_value)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 48, in represent_data
  | node = self.yaml_representers[data_types[0]](self, data)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 199, in represent_list
  | return self.represent_sequence('tag:yaml.org,2002:seq', data)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 92, in represent_sequence
  | node_item = self.represent_data(item)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 58, in represent_data
  | node = self.yaml_representers[None](self, data)
  | File "/usr/lib/python3.6/site-packages/yaml/representer.py", line 231, in represent_undefined
  | raise RepresenterError("cannot represent an object", data)
  | yaml.representer.RepresenterError: ('cannot represent an object', <Box: {'docker-login#v2.0.1': None, 'server': 'containers.amphoratech.net', 'username': 'myUser', 'password-env': 'MY_PASS'}>)
```

Probably related to this: https://stackoverflow.com/questions/18374276/pyaml-throwing-a-representation-error

**Comment by ksindi** [comment]
Thanks @waeljammal. Will take a look.

**Comment by ksindi** [comment]
@waeljammal try the following:
```
      plugins:
        docker-login#v2.0.1:
          server: containers.amphoratech.net
          username: myUser
          password-env: MY_PASS
```
I use buildpipe with docker-login and haven't had issues.

**Comment by jhinklew** [comment]
This doesn't work for all plugins.

https://github.com/bugcrowd/test-summary-buildkite-plugin

This plugin expects the `inputs` to be an array of hash objects. However, if configured as the author suggests Buildpipe throws an error (the same error in the original post)

```yaml
  - label: annotate
    plugins:
      - bugcrowd/test-summary#v1.11.0:
          inputs:
            - label: eslint
              artifact_path: artifacts/eslint.xml
              type: checkstyle
          formatter:
            type: details
          context: test-summary
```

But  if configured how you suggest: 

```yaml
  - label: annotate
    plugins:
      bugcrowd/test-summary#v1.11.0:
        inputs:
            label: eslint
            artifact_path: artifacts/eslint.xml
            type: checkstyle
        formatter:
          type: details
        context: test-summary
```

The plugin throws the following error

```
...

docker.io/bugcrowd/test-summary-buildkite-plugin:v1.11.0
--
  | /usr/src/app/lib/test_summary_buildkite_plugin/runner.rb:32:in `block in inputs': undefined method `merge' for [:type, "checkstyle"]:Array (NoMethodError)
  | from /usr/src/app/lib/test_summary_buildkite_plugin/runner.rb:32:in `each'
  | from /usr/src/app/lib/test_summary_buildkite_plugin/runner.rb:32:in `map'
  | from /usr/src/app/lib/test_summary_buildkite_plugin/runner.rb:32:in `inputs'
  | from /usr/src/app/lib/test_summary_buildkite_plugin/runner.rb:16:in `run'
  | from /usr/src/app/lib/test_summary_buildkite_plugin.rb:21:in `run'
  | from ./bin/run:10:in `<main>'
  | Command failed with exit status: 1
....
```

I've tried using yaml anchors to circumvent the issue but the Buildpipe schema doesn't allow additional properties to be defined.

Kinda stuck between a rock and a hard place 😞 Is there something else I could try/do? We really like Buildpipe and would like to keep using it along with other Buildkite plugins 🙏

**Comment by ksindi** [comment]
Thanks @jhinklew. Glad you're using it. Can you please confirm you're using the latest version of buildpipe?

**Comment by ksindi** [comment]
Feel free to also make a PR btw. More than welcome to.

**Comment by jhinklew** [comment]
```
[ec2-user@ip-10-103-10-34 ~]$ buildpipe --version
0.5.2
```

We have a bootstrap script that runs `sudo pip3 install buildpipe` when a new BK agent node is launched.

**Comment by ksindi** [comment]
Thanks @jhinklew. Let me have a think in the next week or so.

**Comment by ksindi** [comment]
@waeljammal @jhinklew this PR should fix the issue: https://github.com/jwplayer/buildpipe/pull/47

**Comment by ksindi** [comment]
@waeljammal @jhinklew fix has been released as v0.6.0
